src/lib/core/fiber.c

```diff
@@ -893,14 +892,96 @@ fiber_recycle(struct fiber *fiber)
 	fiber->storage.lua.fid_ref = FIBER_LUA_NOREF;
 	unregister_fid(fiber);
 	fiber->fid = 0;
+	/* Set before free to disable truncation system area check. */
+	fiber->gc_initial_size = 0;
 	region_free(&fiber->gc);
+#ifdef ENABLE_BACKTRACE
+	fiber->parent_bt = NULL;
+	fiber->first_alloc_bt = NULL;
+	region_set_callbacks(&fiber->gc, NULL, NULL, NULL);
+#endif /* ENABLE_BACKTRACE */
 	if (!has_custom_stack) {
 		rlist_move_entry(&cord()->dead, fiber, link);
 	} else {
 		cord_add_garbage(cord(), fiber);
 	}
 }
 
+#ifdef ENABLE_BACKTRACE
+/**
+ * Called on allocation on region gc. Saves allocation caller backtrace
+ * to report it later if region gc leak is found.
+ */
+static void
+fiber_on_gc_alloc(struct region *region,
+		  size_t size,
+		  void *opaque)
```

Comment:
Nit: Usually, we don't place every argument on its own line:

<details><summary>diff</summary>

```diff
diff --git a/src/lib/core/fiber.c b/src/lib/core/fiber.c
index 8ca22cb845d0..6c9ad05c8a56 100644
--- a/src/lib/core/fiber.c
+++ b/src/lib/core/fiber.c
@@ -913,9 +913,7 @@ fiber_recycle(struct fiber *fiber)
  * to report it later if region gc leak is found.
  */
 static void
-fiber_on_gc_alloc(struct region *region,
-		  size_t size,
-		  void *opaque)
+fiber_on_gc_alloc(struct region *region, size_t size, void *opaque)
 {
 	(void)region;
 	(void)size;
```

</details>

Reply:
Ok. Used proper style.